test_rmw_implementation/test/test_wait_set.cpp

```diff
@@ -190,94 +237,157 @@ TEST_F(CLASSNAME(TestWaitSet, RMW_IMPLEMENTATION), rmw_wait)
+  EXPECT_EQ(nullptr, guard_conditions.guard_conditions[0]);
+  EXPECT_EQ(nullptr, services.services[0]);
+  EXPECT_EQ(nullptr, clients.clients[0]);
   rmw_reset_error();
 
   // Used a valid wait_set, subscription, guard_conditions, services, clients and events with 100ms
   // timeout
+  subscriptions.subscribers[0] = sub->data;
+  guard_conditions.guard_conditions[0] = gc->data;
+  services.services[0] = srv->data;
+  clients.clients[0] = client->data;
+  events.events[0] = &event;
   ret = rmw_wait(
-    subscriptions, guard_conditions, services, clients, events, wait_set,
+    &subscriptions, &guard_conditions, &services, &clients, &events, wait_set,
     &timeout_argument);
   EXPECT_EQ(ret, RMW_RET_TIMEOUT) << rcutils_get_error_string().str;
+  EXPECT_EQ(nullptr, subscriptions.subscribers[0]);
+  EXPECT_EQ(nullptr, guard_conditions.guard_conditions[0]);
+  EXPECT_EQ(nullptr, services.services[0]);
+  EXPECT_EQ(nullptr, clients.clients[0]);
+  EXPECT_EQ(nullptr, events.events[0]);
   rmw_reset_error();
 
   // Used a valid wait_set, subscription, guard_conditions, services, clients and events with no
   // timeout
+  subscriptions.subscribers[0] = sub->data;
+  guard_conditions.guard_conditions[0] = gc->data;
+  services.services[0] = srv->data;
+  clients.clients[0] = client->data;
+  events.events[0] = &event;
   ret = rmw_wait(
-    subscriptions, guard_conditions, services, clients, events, wait_set,
+    &subscriptions, &guard_conditions, &services, &clients, &events, wait_set,
     &timeout_argument_zero);
   EXPECT_EQ(ret, RMW_RET_TIMEOUT) << rcutils_get_error_string().str;
+  EXPECT_EQ(nullptr, subscriptions.subscribers[0]);
+  EXPECT_EQ(nullptr, guard_conditions.guard_conditions[0]);
+  EXPECT_EQ(nullptr, services.services[0]);
+  EXPECT_EQ(nullptr, clients.clients[0]);
+  EXPECT_EQ(nullptr, events.events[0]);
   rmw_reset_error();
 
-  const char * implementation_identifier = wait_set->implementation_identifier;
-  wait_set->implementation_identifier = "not-an-rmw-implementation-identifier";
-  ret = rmw_wait(
-    subscriptions, guard_conditions, services, clients, events, wait_set,
-    &timeout_argument);
-  wait_set->implementation_identifier = implementation_identifier;
-  EXPECT_EQ(ret, RMW_RET_INCORRECT_RMW_IMPLEMENTATION) << rcutils_get_error_string().str;
-  rmw_reset_error();
-
-
   // Battle test rmw_wait.
   RCUTILS_FAULT_INJECTION_TEST(
   {
+    subscriptions.subscribers[0] = sub->data;
+    guard_conditions.guard_conditions[0] = gc->data;
+    services.services[0] = srv->data;
+    clients.clients[0] = client->data;
+    events.events[0] = &event;
     ret = rmw_wait(
-      subscriptions, guard_conditions, services, clients, events, wait_set,
+      &subscriptions, &guard_conditions, &services, &clients, &events, wait_set,
       &timeout_argument);
 
-    EXPECT_TRUE(RMW_RET_TIMEOUT == ret || RMW_RET_ERROR == ret);
-    rmw_reset_error();
+    if (RMW_RET_TIMEOUT == ret) {
```

Comment:
If `rmw_wait` times out, is it granted that all 5 parameters are going to be already received aka nullptrs?

Reply:
A `nullptr` means they have not been received. I expect none to be activated.

Reply:
Oh, i mised those, then its fine